tests/drivers/charger/sbs_charger/testcase.yaml

```diff
@@ -0,0 +1,30 @@
+tests:
+  # section.subsection
+  drivers.sbs_charger.emulated:
+    tags: drivers charger
+    filter: >
+      dt_compat_enabled("sbs,sbs-charger") and
+      (CONFIG_QEMU_TARGET or CONFIG_BOARD_NATIVE_POSIX)
+    extra_args:
+      CONF_FILE="prj.conf;boards/emulated_board.conf"
+      DTC_OVERLAY_FILE="boards/emulated_board.overlay"
+    platform_exclude:
+      qemu_cortex_a53
+      qemu_cortex_a53_smp
+      qemu_kvm_arm64
+      xenvm
+      xenvm_gicv3
+  drivers.sbs_charger.emulated_64_bit_i2c_addr:
+    tags: drivers charger
+    filter: >
+      dt_compat_enabled("sbs,sbs-charger") and
+      (CONFIG_QEMU_TARGET or CONFIG_BOARD_NATIVE_POSIX)
+    platform_allow:
+      qemu_cortex_a53
+      qemu_cortex_a53_smp
+      qemu_kvm_arm64
+      xenvm
+      xenvm_gicv3
```

Comment:
this looks too complicated, can you please explain why all these QEMU settings here?

Reply:
This is repeating a pattern from https://github.com/zephyrproject-rtos/zephyr/pull/55103.

There's `drivers.sbs_charger.emulated_64_bit_i2c_addr` and `drivers.sbs_charger.emulated`. Some qemu's require 64 bit i2c addresses (in the `reg` field in the overlay) due to the `#address-cells` and `#size-cells` values.

I meant to make a zephyr issue (I forgot) that allows us to filter tests based off those two properties as it'd remove the need for this boilerplate to allow a test to run all the qemus.